modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/PartitionReservationManager.java

```diff
@@ -0,0 +1,443 @@
+                                "(partition of PARTITIONED cache cannot be reserved) [" +
+                                "localNodeId=%s, rmtNodeId=%s, %s, affTopVer=%s, cacheId=%s, " +
+                                "cacheName=%s, part=%s, partFound=%s, partState=%s]",
+                                ctx.localNodeId(),
+                                nodeId,
+                                qryInfo,
+                                topVer,
+                                cctx.cacheId(),
+                                cctx.name(),
+                                partId,
+                                true,
+                                partState
+                            );
+                        }
+
+                        reserved.add(part);
+
+                        reservedCnt++;
+                    }
+                }
+                finally {
+                    top.readUnlock();
+                }
+
+                MTC.span().addLog(() -> "Cache partitions were reserved [cache=" + cctx.name() +
+                    ", partitions=" + partIds + ", topology=" + topVer + ']');
+
+                if (explicitParts == null && reservedCnt > 0) {
+                    // We reserved all the primary partitions for cache, attempt to add group reservation.
+                    GridDhtPartitionsReservation grp = new GridDhtPartitionsReservation(topVer, cctx, "SQL");
+
+                    synchronized (this) {
+                        // Double check under lock.
+                        GridReservable grpReservation = reservations.get(grpKey);
+
+                        if (grpReservation != null)
+                            return groupPartitionReservation(reserved, grpReservation, cctx, topVer, nodeId, qryInfo);
+                        else {
+                            if (grp.register(reserved.subList(reserved.size() - reservedCnt, reserved.size()))) {
+                                reservations.put(grpKey, grp);
+
+                                grp.onPublish(new CI1<>() {
+                                    @Override public void apply(GridDhtPartitionsReservation r) {
+                                        reservations.remove(grpKey, r);
+                                    }
+                                });
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @param cacheName Cache name.
+     */
+    public void onCacheStop(String cacheName) {
+        // Drop group reservations.
+        for (PartitionReservationKey grpKey : reservations.keySet()) {
+            if (F.eq(grpKey.cacheName(), cacheName))
+                reservations.remove(grpKey);
+        }
+    }
+
+    /**
+     * @param cctx Cache context.
+     * @param part Partition.
+     */
+    private static void failQueryOnLostData(
```

Comment:
Trivial. Codestyle. Entire signature might be placed on single line.

Reply:
It can't be placed with exception